cmd/habitat/data_proxy/data_proxy_test.go

```diff
@@ -0,0 +1,113 @@
+package dataproxy
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/eagraf/habitat/cmd/sources"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const rawid = "https://json-schema.org/learn/examples/geographical-location.schema.json"
+
+var geoSch = `
+{
+	"$id": "https://json-schema.org/learn/examples/geographical-location.schema.json",
+	"$schema": "https://json-schema.org/draft/2020-12/schema",
+	"title": "Longitude and Latitude Values",
+	"description": "A geographical coordinate.",
+	"required": [ "latitude", "longitude" ],
+	"type": "object",
+	"properties": {
+	  "latitude": {
+		"type": "number",
+		"minimum": -90,
+		"maximum": 90
+	  },
+	  "longitude": {
+		"type": "number",
+		"minimum": -180,
+		"maximum": 180
+	  }
+	}
+  }`
+
+var id = sources.EncodeId(rawid)
+var geoSchema = &sources.Schema{
+	Schema:      []byte(geoSch),
+	B64id:       id,
+	Name:        "geography",
+	Description: "test json schema",
+}
+
+func TestSourcesWriteRead(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	p := NewDataProxy(ctx, map[string]*DataServerNode{})
+
+	path := "tmp"
+	os.MkdirAll("tmp", os.ModePerm)
```

Comment:
You can use the testing libraries temp directory facilities here (t.TempDir), which should do cleanup automatically.